src/ui/shared/openUrl.ts

```diff
@@ -0,0 +1,37 @@
+import browser from 'webextension-polyfill';
+import {
+  type WindowLayout,
+  type WindowType,
+  WindowParam,
+} from './WindowContext';
+
+interface WindowParams {
+  windowType?: WindowType;
+  windowLayout?: WindowLayout;
+}
+
+export async function openUrl(url: URL, params?: WindowParams) {
+  if (params?.windowType) {
+    url.searchParams.set(WindowParam.windowType, params.windowType);
+  }
+  if (params?.windowLayout) {
+    url.searchParams.set(WindowParam.windowLayout, params.windowLayout);
+  }
+  const [activeTab] = await browser.tabs.query({
+    active: true,
+    lastFocusedWindow: true,
+  });
+  browser.tabs.create({
+    url: url.toString(),
+    index: activeTab?.index + 1,
+  });
+}
+
+export function openHref(event: React.MouseEvent, params?: WindowParams) {
```

Comment:
May be it is better to call this `openExternalHref`? To avoid potential usage for links inside extension?

Reply:
Ah, I see. Well, probably this is ok then

Reply:
Yeah, it is for internal links :)